### ISSUE TITLE
Create tree compare page for comparing two changesets/patchsets each consisting multiple commits.

### ISSUE THREAD

**Issue by ghost** [go-gitea/gitea#12800]
The current diff page only shows how the content tracked by the repo changes between two commits.

Some people would like to do PR review not just reviewing the content of the change but also the fact of how it is structured into commits. To be able to track PR evolution it would be useful to see how the change is moving between commits for changesets/PR revisions. (The reason why some people want this is mostly that they have the philosophy that they are trying to keep a tidy/clean history. Also they are maybe trying to  make "atomic changes" which are  self contained enough to easily move between branches, by e.g.: cherry-picking )

How this should look like visually is an interesting question.

Basically you want to continue to see the change contained in the changesets but also there is an extra additional "dimension" about comparing how they are distributed between commits how they moved between commits. Also the number of commits can change, some or all could be squashed into one commit. 

New additional changes between changesets (e.g.: revisions of a PR) could come in either as brand new separate commits or as part of a previously existing commit.

I am not aware of any system that have UI for something like this.
